src/prefect/engine/executors/dask.py

```diff
@@ -276,31 +279,52 @@ def wait(self, futures: Any) -> Any:
 
 class LocalDaskExecutor(Executor):
     """
-    An executor that runs all functions locally using `dask` and a configurable dask scheduler.  Note that
-    this executor is known to occasionally run tasks twice when using multi-level mapping.
+    An executor that runs all functions locally using `dask` and a configurable
+    dask scheduler.
 
     Args:
-        - scheduler (str): The local dask scheduler to use; common options are "synchronous", "threads" and "processes".  Defaults to "threads".
+        - scheduler (str): The local dask scheduler to use; common options are
+            "threads", "processes", and "synchronous".  Defaults to "threads".
         - **kwargs (Any): Additional keyword arguments to pass to dask config
     """
 
     def __init__(self, scheduler: str = "threads", **kwargs: Any):
+        self._callback = None
         self.scheduler = scheduler
-        self.kwargs = kwargs
+        self.dask_config = kwargs
         super().__init__()
 
+    def __getstate__(self) -> dict:
+        state = self.__dict__.copy()
+        state["_callback"] = None
+        return state
+
+    def __setstate__(self, state: dict) -> None:
+        self.__dict__.update(state)
+
     @contextmanager
     def start(self) -> Iterator:
-        """
-        Context manager for initializing execution.
-
-        Configures `dask` and yields the `dask.config` contextmanager.
-        """
+        """Context manager for initializing execution."""
         # import dask here to reduce prefect import times
-        import dask
+        from dask.callbacks import Callback
+
+        class PrefectCallback(Callback):
+            def __init__(self):  # type: ignore
+                self.cache = {}
+
+            def _start(self, dsk):  # type: ignore
```

Comment:
So is `_start` called at the beginning of computation for every task?

Reply:
I guess I'd be curious to learn about how this works exactly, I don't think I fully understand it.  (whether here or offline)

Reply:
Here's the docs on callbacks: https://docs.dask.org/en/latest/diagnostics-local.html#custom-callbacks, does this help? The callback here lets us modify the graph before computation starts - `_start` is called once on `compute()` setup, `_posttask` is called after every task.

Reply:
I did read those docs but didn't realize that `_start` was called on the first call to `compute` when the graph is fully known (I couldn't quite tell what "state" was referring to, but that might be because my brain is in `distributed` mode and I forgot no work occurs until `compute` is called).

I guess I'm still confused though because I don't see how `self.cache` is populated at this stage?  From reading that doc on `_posttask` as well I assumed it wouldn't be populated until after the task finishes (which necessary occurs _after_ compute has been called).

Reply:
Oh I think I'm being silly - this callback protects us against multiple calls to `compute`, not reruns within a single `compute` call 👍

Reply:
Yah. Only a single `cache` exists during the lifetime of a `Executor.start()` context, and is built up over repeated `compute` calls with (actually `get` calls, which is one level lower). The `_start` callback lets us swap out already computed tasks for their results, preventing them from being recomputed. The `_posttask` callback adds the results of complete tasks to the `cache` to be reused later.

Reply:
Awesome thank you!